### ISSUE TITLE
Proposal: Add support for compiling with HDF5 v1.12

### ISSUE THREAD

**Issue by slydex** [openmc-dev/openmc#1542]
Due to changes in the most recent version 1.12 of HDF5, i had to add 
```
H5Oget_info_vers=1 
H5O_info_t_vers=1 
H5Oget_info_by_idx_vers=1
```

as CXXFLAGS in order for openmc to compile. E.g.:
```
CXXFLAGS=-DH5Oget_info_vers=1 -DH5O_info_t_vers=1 -DH5Oget_info_by_idx_vers=1 cmake ..
```
It would be great to have a HDF5 version check added to the CMakeLists.txt .
Sadly i didn't find an cmake-style way to do it as the `HDF5_VERSION` cmake variable seems unused (at least in my implementation). The easiest solution i see is to have a separate openmc specific flag as for e.g. `-Doptimize`.

**Comment by jtramm** [comment]
This should be fixed in #1533 -- but let us know if it's still not working for you.

**Comment by paulromano** [comment]
Closing this, but feel free to reopen @slydex if you're still having issues
